src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -149,19 +162,116 @@ public int getTotalMessageCount(){
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
   
-  /* Returns all laptops */
-  public List<Laptop> getAllLaptops(){
-	 List<Laptop> laptops = new ArrayList<>();
-	 laptops.add(new Laptop("Dell","Black","Windows",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",13,1200.00));
-	 laptops.add(new Laptop("Dell","Grey","Windows",15,800.00));
-	 laptops.add(new Laptop("Dell","Black","Linux",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",15,1699.00));
-	 laptops.add(new Laptop("Macbook Pro","Black","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Black","Mac",15,1900.00));
-	 laptops.add(new Laptop("Macbook Pro","Grey","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Grey","Mac",15,1900.00));
-
-	 return laptops;
-  }
-}
\ No newline at end of file
+  /**
+   * @return list of all Laptops
+   */
+  public List<Laptop> getAlllaptopList(){
+	 return laptopList;
+  }
+  
+  /** Adds a laptop to the LaptopList
+   * @param Laptop to add to the list 
+   */
+  public void addLaptop(Laptop newLaptop) {
+	  laptopList.add(newLaptop);
+  }
+  
+  /**Returns a laptop list with laptops of a certain brand*/
+  public List<Laptop> searchBrand(String brand){
+	  //list of laptops that meet the brand requirements 
+	  List<Laptop> brandreturn=new ArrayList<>();
+	  Iterator<Laptop> it= laptopList.iterator(); 
+	  while(it.hasNext()) {
+		  if(it.next().getBrand()==brand.toLowerCase())
```

Comment:
You almost never want to use `==` to compare `String` values. Instead, you probably want to use the `.equals()` function. More info here: https://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java

(You might also want to think about whether it's possible for a `brand` to be null.)

Reply:
(Same thing in some of the functions below.)